app/controllers/shipit/api/stacks_controller.rb

```diff
@@ -78,6 +84,37 @@ def stack
         @stack ||= stacks.from_param!(params[:id])
       end
 
+      def update_lock
+        if key?(:lock_reason)
+          reason = params[:lock_reason]
+          if reason.present?
+            stack.lock(reason, nil)
+          elsif stack.locked?
+            stack.unlock
+          end
+        end
+      end
+
+      def update_archived
+        if key?(:archived)
+          if params[:archived] == "true"
```

Comment:
This doesn't seem right at all. With `explicit-parameters` you get proper types. `:archived` is declared as a boolean, you should compare it with a String.

Reply:
Sorry, my bad, I should keep them consistent

Reply:
Done

Reply:
This should be a boolean. The API is always JSON, so it makes no sense to pass it as a string.

Reply:
That's my fault, the API tests should all pass `as: :json` to better simulate true API requests. I fixed that in https://github.com/Shopify/shipit-engine/pull/1283, if you rebase you should be able to test with true booleans.